### PR TITLE
feat: run goose in a docker-style sandbox

### PR DESCRIPTION
**Pull request by michaelneale**
This change allows goose to run inside a docker container for those you are more interested in just trying it out without it running entirely in their environment. 

To build it: 

```sh
docker build -t goose . 
docker run -it --env OPENAI_API_KEY goose
```

This will open a new goose session that uses OPENAI (can do similar with anthropic etc) in an environment you can let it play in (it can clone things, try to build etc) all in a container. 

If interested, can have this automatically published to make it easier for people to tyre kick

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+RUN pipx install goose-ai
+
+# Make sure the PATH is updated
+ENV PATH="/root/.local/bin:${PATH}"
+
+# Run an infinite loop to keep the container running for testing
+ENTRYPOINT ["goose", "session", "start"]
+
+# once built, you can run this with something like: 
+#    docker run -it --env OPENAI_API_KEY goose-ai
```
Comment (codefromthecrypt) [inline Dockerfile:23]: Maybe include an example for local?

```suggestion
#    docker run -it --env OPENAI_API_KEY goose-ai
# or to run against ollama running on the same host
#    docker run -it --env OPENAI_HOST=http://host.docker.internal:11434/ --env OPENAI_API_KEY=unused goose-ai
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+WORKDIR /app
+
+# Install SSL certificates, update certifi, install pipx and move to path
+RUN apt-get update && apt-get install -y ca-certificates git curl make \
+    && pip install --upgrade certifi \
+    && pip install pipx \
+    && pipx ensurepath
+
+# Install goose-ai CLI using pipx
+RUN pipx install goose-ai
```
Comment (lifeizhou-ap) [inline Dockerfile:14]: If the image was built before and this layer is cached, the user won't have the latest `goose-ai` if there is a new release. 

We could either to disable the cache of this layer  or the user can install the latest in the docker container

Comment (michaelneale) [reply Dockerfile:14]: @lifeizhou-ap what about if this instead was built as part of the py release workflow - and pushed as a versioned image to the GH docker registry - would that be suitable?

**Review by baxen — APPROVED** [review]
Looks good! We'll revisit publishing these in the future

**Pull request merged**